design-proposals/eim-standalone-managed-en.md

```diff
@@ -0,0 +1,236 @@
+- Enabling the onboarding process at scale, i.e., the design should enable onboarding of multiple machines at once.
+
+The proposed solution is an MVP approach - some of the above goals may not be fully achieved in the first version of the design,
+but the design doesn't preclude achieving them in the future. 
+
+### Assumptions
+
+- Customers will drive the onboarding process from a local developer machine, with desktop, keyboard and mouse.
+- The local developer machine will have direct access to ENs via local subnet. Customers can SSH into the ENs.
+- SENs being onboarded won't be equipped with desktop, keyboard or mouse. The only machine equipped with peripherals will be
+  the local developer/admin machine.
+- The design will re-use the current APIs of Onboarding Manager to drive IO/NIO-based onboarding.
+  We may just require small modifications to the Onboarding Manager and the gRPC interface.
+
+### Solution
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+
+  box LightYellow Edge Node(s)
+  participant ctl as Standalone Onboarding CLI
+  end
+
+  participant user as User
+
+  box rgb(235,255,255) Edge Orchestrator
+
+  participant pa as Provisioning Nginx
+  participant om as Onboarding Manager
+  participant kc as KeyCloak
+  participant inv as Inventory / API
+
+  end
+
+  note over ctl,user: User installs Standalone EN, decides to scale out
+
+  note over pa,inv: OS profiles created beforehand, no Host/Instance pre-registration
+
+  user->>user: Retrieves standalone onboarding user credentials 
+  user->>ctl: Logs in to the EN and invokes CLI tool
+  
+  note over ctl: CLI triggered
+
+  ctl->>ctl: Checks local OS prerequisites
+
+  ctl->>ctl: Take user inputs and validate
+
+  ctl->>+pa: Get orchestrator CA certificate
+  pa-->>-ctl: [CA certificate]
+
+  ctl->>+kc: Retrieve JWT token for standalone onboarding user
+  kc->>-ctl: [JWT token]
+
+  ctl->>+om: Invoke OnboardStandaloneService gRPC (UUID, Serial Number, OS info, ...)
+
+  om->>om: Validate JWT standalone onboarding role
+
+  om->>inv: Query Inventory for OS resource by OS version and distribution
+
+  om->>om: Generate cloud-init for Standalone EN
+
+  om->>inv: Create Host and Instance resource, Set EN as Onboarded and Provisioned
+  om->>-ctl: Return cloud-init
+
+  ctl->>ctl: Save cloud-init & run manually or reboot
+
+  note over ctl: After OS reboot or manual cloud-init execution
+
+  ctl->>ctl: Activate Edge Node Agents
+```
+
+**Step 0.** A user already provisioned a set of Standalone ENs following the user guides and decides to scale out.
+They don't need to have direct access to the Edge Orchestrator UI/API (could be different personas on-site vs. remote administrator).
+Users don't need to perform any configuration on the Edge Orchestrator beforehand, but we assume that the Edge Orchestrator supports
+the OS version of the Standalone ENs. The remote administrator configures a special user/role for SEN onboarding.
```

Comment:
Is there an assumption that the "user/role for SEN" is associated with a single project that will determine in which tenant the SEN is imported?

Reply:
Yes, the same assumption we have now for IO onboarding

Reply:
I believe we want just separate from the host/onboarding roles